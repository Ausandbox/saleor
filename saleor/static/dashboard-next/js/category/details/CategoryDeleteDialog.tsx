import Button from "material-ui/Button";
import Dialog, {
  DialogActions,
  DialogContent,
  DialogTitle
} from "material-ui/Dialog";
import Typography from "material-ui/Typography";
import { withStyles } from "material-ui/styles";
import * as React from "react";

import { pgettext } from "../../i18n";

interface CategoryDeleteDialogProps {
  onClose?();
  onConfirm?();
  opened?: boolean;
  title: string;
}

export const CategoryDeleteDialog: React.StatelessComponent<
  CategoryDeleteDialogProps
> = props => {
  const { title, children, opened, onConfirm, onClose, ...dialogProps } = props;
  return (
    <Dialog open={opened} {...dialogProps}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>{children}</DialogContent>
      <DialogActions>
        <Button color="primary" onClick={onClose}>
          {pgettext("Dashboard cancel action", "Cancel")}
        </Button>
        <Button color="primary" variant="raised" onClick={onConfirm}>
          {pgettext("Dashboard delete action", "Remove")}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
