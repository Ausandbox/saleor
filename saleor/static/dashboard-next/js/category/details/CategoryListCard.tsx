import * as React from "react";
import Typography from "material-ui/Typography";
import { Link } from "react-router-dom";
import { withStyles } from "material-ui/styles";

import { gettext } from "../../i18n";
import { Button, Toolbar, Paper } from "material-ui";

const decorate = withStyles(theme => ({
  spacer: {
    flex: "1 1 100%"
  },
  actions: {
    whiteSpace: "nowrap",
    minWidth: "auto"
  },
  title: {
    flex: "0 0 auto"
  }
}));

interface ListCardProps {
  addActionLabel: string;
  addActionLink: string;
  label: string;
}

export const CategoryListCard = decorate<ListCardProps>(
  ({ addActionLabel, addActionLink, children, classes, label }) => (
    <Paper>
      <Toolbar>
        <div className={classes.title}>
          <Typography variant="title">{label}</Typography>
        </div>
        <div className={classes.spacer} />
        <Button
          className={classes.actions}
          color="primary"
          component={props => <Link to={addActionLink} {...props} />}
        >
          {addActionLabel}
        </Button>
      </Toolbar>
      {children}
    </Paper>
  )
);

export default CategoryListCard;
