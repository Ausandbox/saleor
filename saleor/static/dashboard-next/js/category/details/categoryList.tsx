import * as React from "react";
import { Component } from "react";
import { graphql } from "react-apollo";

import { ListCard } from "../../components/cards";
import { Navigator } from "../../components/Navigator";
import { Skeleton } from "../../components/Skeleton";
import { Table } from "../../components/Table";
import { TableRow, TableCell } from "material-ui";
import { categoryAddUrl, categoryShowUrl } from "../index";
import { categoryChildren, rootCategoryChildren } from "../queries";
import { gettext, pgettext } from "../../i18n";

const tableHeaders = [
  {
    label: pgettext("Category list table header name", "Name"),
    name: "name"
  },
  {
    label: pgettext("Category list table header description", "Description"),
    name: "description",
    wide: true
  }
];
const PAGINATE_BY = 4;

const feederOptions = {
  options: props => ({
    variables: {
      id: props.categoryId,
      first: props.filters.after
        ? PAGINATE_BY
        : props.filters.before ? null : PAGINATE_BY,
      last: props.filters.before ? PAGINATE_BY : null,
      before: props.filters.before,
      after: props.filters.after
    },
    fetchPolicy: "network-only"
  })
};
const categoryListFeeder = graphql(categoryChildren, feederOptions);
const rootCategoryListFeeder = graphql(rootCategoryChildren, feederOptions);

interface BaseCategoryListProps {
  categoryId?: string;
  categories: {
    edges: Array<any>;
    pageInfo: {
      hasNextPage: boolean;
      hasPreviousPage: boolean;
      startCursor: string;
      endCursor: string;
    };
  };
  label: string;
  loading: boolean;
  filters: {
    first?: string;
    last?: string;
    after?: string;
    before?: string;
  };
}

class BaseCategoryList extends Component<BaseCategoryListProps> {
  render() {
    const { label, loading, categories, filters, categoryId } = this.props;
    const pageInfo = {
      hasPreviousPage: filters.after
        ? true
        : categories.pageInfo.hasPreviousPage,
      hasNextPage: filters.before ? true : categories.pageInfo.hasNextPage
    };
    return (
      <Navigator>
        {navigate => (
          <ListCard
            addActionLabel={gettext("Create category")}
            addActionLink={categoryAddUrl(categoryId)}
            label={label}
          >
            <Table
              headers={tableHeaders}
              onNextPage={() =>
                navigate(`?after=${categories.pageInfo.endCursor}`)
              }
              onPreviousPage={() =>
                navigate(`?before=${categories.pageInfo.startCursor}`)
              }
              page={pageInfo}
            >
              {loading ? (
                <TableRow>
                  {tableHeaders.map(header => (
                    <TableCell key={header.name}>
                      <Skeleton style={{ width: "80%" }} />
                    </TableCell>
                  ))}
                </TableRow>
              ) : (
                categories.edges.map(({ node }) => (
                  <TableRow
                    key={node.id}
                    onClick={() => navigate(categoryShowUrl(node.id))}
                  >
                    {tableHeaders.map(header => (
                      <TableCell key={header.name}>
                        {node[header.name]}
                      </TableCell>
                    ))}
                  </TableRow>
                ))
              )}
            </Table>
          </ListCard>
        )}
      </Navigator>
    );
  }
}

interface CategoryListComponentProps {
  categoryId: string;
  data: {
    loading: boolean;
    category: {
      children: {
        edges: Array<any>;
        pageInfo: {
          hasNextPage: boolean;
          hasPreviousPage: boolean;
          startCursor: string;
          endCursor: string;
        };
      };
    };
  };
  filters: {
    first?: string;
    last?: string;
    after?: string;
    before?: string;
  };
}

const CategoryListComponent: React.StatelessComponent<
  CategoryListComponentProps
> = props => {
  const { data } = props;
  const categories = data.category
    ? data.category.children
    : {
        edges: [],
        pageInfo: {
          hasNextPage: false,
          hasPreviousPage: false,
          startCursor: "",
          endCursor: ""
        }
      };
  return (
    <BaseCategoryList
          categories={categories}
          label={pgettext("Title of the subcategories list", "Subcategories")}
          loading={data.loading}
          filters={props.filters}
        categoryId={props.categoryId}/>

  );
};
const CategoryList = categoryListFeeder(CategoryListComponent);

interface RootCategoryListComponentProps {
  data: {
    loading: boolean;
    categories: {
      edges: Array<any>;
      pageInfo: {
        hasNextPage: boolean;
        hasPreviousPage: boolean;
        startCursor: string;
        endCursor: string;
      };
    };
  };
  filters: {
    first?: string;
    last?: string;
    after?: string;
    before?: string;
  };
}

const RootCategoryListComponent: React.StatelessComponent<
  RootCategoryListComponentProps
> = props => {
  const { data } = props;
  const categories = data.categories || {
    edges: [],
    pageInfo: {
      hasNextPage: false,
      hasPreviousPage: false,
      startCursor: "",
      endCursor: ""
    }
  };
  return (
    <BaseCategoryList
      categories={categories}
      label={pgettext("Title of the categories list", "Categories")}
      loading={data.loading}
      filters={props.filters}
    />
  );
};
const RootCategoryList = rootCategoryListFeeder(RootCategoryListComponent);

export { CategoryList, RootCategoryList };
