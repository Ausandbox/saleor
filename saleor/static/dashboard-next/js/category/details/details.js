import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { graphql, compose } from "react-apollo";
import { withRouter } from "react-router-dom";

import { ConfirmRemoval } from "../../components/modals";
import { DescriptionCard } from "../../components/cards";
import { categoryDelete } from "../mutations";
import { categoryDetails } from "../queries";

const categoryDetailsQueryFeeder = graphql(categoryDetails, {
  options: props => ({
    variables: {
      id: props.categoryId
    }
  })
});
const categoryRemoveMutationFeeder = graphql(categoryDelete);

@withRouter
@compose(categoryDetailsQueryFeeder, categoryRemoveMutationFeeder)
class CategoryProperties extends Component {
  static propTypes = {
    categoryId: PropTypes.string,
    data: PropTypes.shape({
      category: PropTypes.shape({
        parent: PropTypes.shape({
          id: PropTypes.string
        })
      })
    }),
    history: PropTypes.object,
    mutate: PropTypes.func
  };

  constructor(props) {
    super(props);
    this.handleRemoveButtonClick = this.handleRemoveButtonClick.bind(this);
    this.handleRemoveAction = this.handleRemoveAction.bind(this);
    this.state = { opened: false };
  }

  handleEditButtonClick = () => {
    this.props.history.push(`/categories/${this.props.categoryId}/edit`);
  };

  handleRemoveButtonClick() {
    this.setState(prevState => ({ opened: !prevState.opened }));
  }

  handleRemoveAction() {
    this.props
      .mutate({
        variables: {
          id: this.props.categoryId
        }
      })
      .then(() => {
        this.handleRemoveButtonClick();
        this.props.history.replace(
          `/categories/${
            this.props.data.category.parent
              ? this.props.data.category.parent.id
              : ""
          }/`
        );
      });
  }

  render() {
    const { data } = this.props;
    const titleFmt = pgettext(
      "Remove category modal title",
      "Remove category %s"
    );
    const contentFmt = pgettext(
      "Remove category modal title",
      "Are you sure you want to remove category <strong>%s</strong>?"
    );
    return (
      <div>
        {data.loading ? (
          <span>loading</span>
        ) : (
          <Fragment>
            <DescriptionCard
              description={data.category.description}
              handleEditButtonClick={this.handleEditButtonClick}
              handleRemoveButtonClick={this.handleRemoveButtonClick}
              title={data.category.name}
            />
            <ConfirmRemoval
              onClose={this.handleRemoveButtonClick}
              onConfirm={this.handleRemoveAction}
              opened={this.state.opened}
              title={interpolate(titleFmt, [data.category.name])}
            >
              <p
                dangerouslySetInnerHTML={{
                  __html: interpolate(contentFmt, [data.category.name])
                }}
              />
              {data.category.products &&
                data.category.products.totalCount > 0 && (
                  <p>
                    {interpolate(
                      ngettext(
                        "There is one product in this category that will be removed.",
                        "There are %s products in this category that will be removed.",
                        data.category.products.totalCount
                      ),
                      [data.category.products.totalCount]
                    )}
                  </p>
                )}
            </ConfirmRemoval>
          </Fragment>
        )}
      </div>
    );
  }
}

export default CategoryProperties;
