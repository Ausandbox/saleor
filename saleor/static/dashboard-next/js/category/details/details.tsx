import * as React from "react";
import { Component } from "react";
import { Mutation, Query } from "react-apollo";

import { ConfirmRemoval } from "../../components/modals";
import { DescriptionCard } from "../../components/cards";
import { Navigator } from "../../components/Navigator";
import { categoryDelete } from "../mutations";
import { categoryDetails } from "../queries";
import { pgettext, interpolate, ngettext } from "../../i18n";
import { categoryEditUrl, categoryShowUrl } from "../index";

interface CategoryPropertiesProps {
  categoryId: string;
}

interface CategoryPropertiesState {
  opened: boolean;
}

class CategoryProperties extends Component<
  CategoryPropertiesProps,
  CategoryPropertiesState
> {
  state = { opened: false };

  handleRemoveButtonClick = () => {
    this.setState(prevState => ({ opened: !prevState.opened }));
  };

  render() {
    const { categoryId } = this.props;
    const titleFmt = pgettext(
      "Remove category modal title",
      "Remove category %s"
    );
    const contentFmt = pgettext(
      "Remove category modal title",
      "Are you sure you want to remove category <strong>%s</strong>?"
    );
    return (
      <Navigator>
        {navigate => (
      <Query query={categoryDetails} variables={{ id: categoryId }}>
        {result => (
          <Mutation mutation={categoryDelete} variables={{ id: categoryId }}>
            {deleteCategory => {
              const { data: { category } } = result;
              const handleRemoveAction = async () => {
                await deleteCategory();
                this.handleRemoveButtonClick;
                navigate(
                  categoryShowUrl(
                    category.parent ? category.parent.url : null
                  ),
                  true
                );
              };

              return (
                <>
                  <DescriptionCard
                    description={result.loading ? "" : category.description}
                    editButtonLink={categoryEditUrl(category.id)}
                    handleRemoveButtonClick={this.handleRemoveButtonClick}
                    loading={result.loading}
                    title={result.loading ? "" : category.name}
                  />
                  {!result.loading ? (
                    <ConfirmRemoval
                      onClose={this.handleRemoveButtonClick}
                      onConfirm={handleRemoveAction}
                      opened={this.state.opened}
                      title={interpolate(titleFmt, [category.name])}
                    >
                      <p
                        dangerouslySetInnerHTML={{
                          __html: interpolate(contentFmt, [category.name])
                        }}
                      />
                      {category.products &&
                        category.products.totalCount > 0 && (
                          <p>
                            {interpolate(
                              ngettext(
                                "There is one product in this category that will be removed.",
                                "There are %s products in this category that will be removed.",
                                category.products.totalCount
                              ),
                              [category.products.totalCount]
                            )}
                          </p>
                        )}
                    </ConfirmRemoval>
                  ) : null}
                </>
              );
            }}
          </Mutation>
        )}
      </Query>
        )}
      </Navigator>
    );
  }
}

export default CategoryProperties;
