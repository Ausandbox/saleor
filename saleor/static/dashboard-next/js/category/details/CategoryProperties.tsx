import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";
import { Component } from "react";
import { Query } from "react-apollo";
import { Redirect } from "react-router-dom";

import { CategoryDeleteDialog } from "./CategoryDeleteDialog";
import { CategoryDetails } from "./CategoryDetails";
import {
  TypedCategoryDeleteMutation,
  categoryDeleteMutation
} from "../mutations";
import { TypedCategoryDetailsQuery, categoryDetailsQuery } from "../queries";
import { pgettext, interpolate, ngettext } from "../../i18n";
import { categoryEditUrl, categoryShowUrl } from "../index";

interface CategoryPropertiesProps {
  categoryId: string;
}

interface CategoryPropertiesState {
  opened: boolean;
}

export class CategoryProperties extends Component<
  CategoryPropertiesProps,
  CategoryPropertiesState
> {
  state = { opened: false };

  handleRemoveButtonClick = () => {
    this.setState(prevState => ({ opened: !prevState.opened }));
  };

  render() {
    const { categoryId } = this.props;
    const titleFmt = pgettext(
      "Remove category modal title",
      "Remove category %s"
    );
    const contentFmt = pgettext(
      "Remove category modal title",
      "Are you sure you want to remove category <strong>%s</strong>?"
    );
    return (
      <TypedCategoryDetailsQuery
        query={categoryDetailsQuery}
        variables={{ id: categoryId }}
      >
        {result => (
          <TypedCategoryDeleteMutation
            mutation={categoryDeleteMutation}
            variables={{ id: categoryId }}
          >
            {(deleteCategory, deleteResult) => {
              const { data: { category } } = result;
              if (deleteResult && !deleteResult.loading) {
                this.handleRemoveButtonClick();
                return (
                  <Redirect
                    to={categoryShowUrl(
                      category.parent ? category.parent.id : null
                    )}
                    push={false}
                  />
                );
              }

              return (
                <>
                  <CategoryDetails
                    description={result.loading ? "" : category.description}
                    editButtonLink={
                      result.loading ? "" : categoryEditUrl(category.id)
                    }
                    handleRemoveButtonClick={this.handleRemoveButtonClick}
                    loading={result.loading}
                    title={result.loading ? "" : category.name}
                  />
                  {!result.loading ? (
                    <CategoryDeleteDialog
                      onClose={this.handleRemoveButtonClick}
                      onConfirm={() => deleteCategory()}
                      opened={this.state.opened}
                      title={interpolate(titleFmt, [category.name])}
                    >
                      <DialogContentText
                        dangerouslySetInnerHTML={{
                          __html: interpolate(contentFmt, [category.name])
                        }}
                      />
                      {category.products &&
                        category.products.totalCount > 0 && (
                          <DialogContentText>
                            {interpolate(
                              ngettext(
                                "There is one product in this category that will be removed.",
                                "There are %s products in this category that will be removed.",
                                category.products.totalCount
                              ),
                              [category.products.totalCount]
                            )}
                          </DialogContentText>
                        )}
                    </CategoryDeleteDialog>
                  ) : null}
                </>
              );
            }}
          </TypedCategoryDeleteMutation>
        )}
      </TypedCategoryDetailsQuery>
    );
  }
}

export default CategoryProperties;
