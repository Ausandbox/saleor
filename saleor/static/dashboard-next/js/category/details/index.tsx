import * as React from "react";
import Grid from "material-ui/Grid";

import { CategoryProperties } from "./CategoryProperties";
import { CategoryList, RootCategoryList } from "./CategoryList";
import { screenSizes } from "../../misc";

interface CategoryDetailsProps {
  filters: any;
  id: string;
}

// TODO: Plug-in filters
const CategoryDetails: React.StatelessComponent<CategoryDetailsProps> = ({
  filters,
  id
}) => (
  <Grid container spacing={24}>
    <Grid item xs={12} md={9}>
      {id ? (
        <Grid container spacing={24}>
          <Grid item xs={12}>
            <CategoryProperties categoryId={id} />
          </Grid>
          <Grid item xs={12}>
            <CategoryList categoryId={id} filters={filters} />
          </Grid>
        </Grid>
      ) : (
        <RootCategoryList filters={filters} />
      )}
    </Grid>
  </Grid>
);

export default CategoryDetails;
