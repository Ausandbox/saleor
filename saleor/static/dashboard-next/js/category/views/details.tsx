import Button from "material-ui/Button";
import Divider from "material-ui/Divider";
import Grid from "material-ui/Grid";
import IconButton from "material-ui/IconButton";
import Toolbar from "material-ui/Toolbar";
import Typography from "material-ui/Typography";
import { withStyles } from "material-ui/styles";
import ArrowBack from "material-ui-icons/ArrowBack";
import * as React from "react";
import { Link } from "react-router-dom";

import {
  categoryPropertiesQuery,
  TypedCategoryPropertiesQuery,
  rootCategoryChildrenQuery,
  TypedRootCategoryChildrenQuery
} from "../queries";
import CategoryProperties from "../components/CategoryProperties";
import CategoryChildElement from "../components/CategoryChildElement";
import CategoryList from "../components/CategoryList";
import ProductChildElement from "../components/ProductChildElement";
import ProductList from "../components/ProductList";
import { categoryShowUrl, categoryAddUrl } from "../index";
import PageHeader from "../../components/PageHeader";
import Skeleton from "../../components/Skeleton";
import i18n from "../../i18n";

const decorate = withStyles(theme => ({
  grid: {
    padding: theme.spacing.unit * 2
  },
  title: {
    flex: 1
  },
  subtitle: {
    display: "flex",
    alignItems: "center" as "center",
    marginBottom: theme.spacing.unit * 2
  }
}));

interface CategoryDetailsProps {
  filters: any;
  id: string;
}

// TODO: Plug-in filters
const CategoryDetails = decorate<CategoryDetailsProps>(
  ({ classes, filters, id }) => {
    if (id) {
      return (
        <TypedCategoryPropertiesQuery
          query={categoryPropertiesQuery}
          variables={{ id, first: 12 }}
          fetchPolicy="network-only"
        >
          {({ loading, error, data: { category }, fetchMore }) => {
            if (error) {
              return <span>not ok</span>;
            }
            const handleLoadMore = () => {
              if (loading) {
                return;
              }
              return fetchMore({
                variables: {
                  after: category.products.pageInfo.endCursor
                },
                updateQuery: (previousResult, { fetchMoreResult }) => {
                  return {
                    ...previousResult,
                    category: {
                      ...previousResult.category,
                      products: {
                        ...previousResult.category.products,
                        edges: [
                          ...previousResult.category.products.edges,
                          ...fetchMoreResult.category.products.edges
                        ],
                        pageInfo: {
                          ...fetchMoreResult.category.products.pageInfo
                        }
                      }
                    }
                  };
                }
              });
            };
            return (
              <>
                <PageHeader
                  backLink={
                    category
                      ? categoryShowUrl(category.parent && category.parent.id)
                      : "#"
                  }
                  loading={loading}
                  title={(category && category.name) || ""}
                />
                <Grid container spacing={24} className={classes.grid}>
                  <Grid item xs={12}>
                    <CategoryProperties category={category} loading={loading} />
                  </Grid>
                  <Grid item xs={12}>
                    <div className={classes.subtitle}>
                      <Typography className={classes.title} variant="title">
                        {i18n.t("Subcategories", { context: "title" })}
                      </Typography>
                      <Button
                        color="secondary"
                        component={props => (
                          <Link
                            to={category ? categoryAddUrl(category.id) : "#"}
                            {...props}
                          />
                        )}
                        disabled={loading}
                        variant="raised"
                      >
                        {i18n.t("Add category", { context: "button" })}
                      </Button>
                    </div>
                    <CategoryList
                      categories={
                        category && category.children && category.children.edges
                      }
                    />
                  </Grid>
                  <Grid item xs={12}>
                    <div className={classes.subtitle}>
                      <Typography className={classes.title} variant="title">
                        {i18n.t("Products", { context: "title" })}
                      </Typography>
                      <Button
                        color="secondary"
                        component={props => <Link to="#" {...props} />}
                        disabled={loading}
                        variant="raised"
                      >
                        {i18n.t("Add product", { context: "button" })}
                      </Button>
                    </div>
                    <ProductList
                      products={
                        category && category.products && category.products.edges
                      }
                      handleLoadMore={handleLoadMore}
                      canLoadMore={
                        category &&
                        category.products &&
                        category.products.pageInfo &&
                        category.products.pageInfo.hasNextPage
                      }
                    />
                  </Grid>
                </Grid>
              </>
            );
          }}
        </TypedCategoryPropertiesQuery>
      );
    }
    return (
      <TypedRootCategoryChildrenQuery
        query={rootCategoryChildrenQuery}
        fetchPolicy="network-only"
      >
        {({ error, loading, data: { categories } }) => {
          if (error) {
            return <span>not ok</span>;
          }
          return (
            <>
              <PageHeader title={i18n.t("Categories", { context: "title" })}>
                <Button
                  color="secondary"
                  component={props => <Link to={categoryAddUrl()} {...props} />}
                  disabled={loading}
                  variant="raised"
                >
                  {i18n.t("Add category", { context: "button" })}
                </Button>
              </PageHeader>
              <div className={classes.grid}>
                <CategoryList categories={categories && categories.edges} />
              </div>
            </>
          );
        }}
      </TypedRootCategoryChildrenQuery>
    );
  }
);

export default CategoryDetails;
