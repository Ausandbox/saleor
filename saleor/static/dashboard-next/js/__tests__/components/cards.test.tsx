import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import { CategoryListCard } from "../../category/details/CategoryListCard";

const listCardProps = {
  displayLabel: true,
  headers: [
    {
      name: "prop1",
      label: "Prop1"
    }
  ],
  list: [
    {
      prop1: "value1",
      prop2: "value2"
    }
  ],
  label: "Title",
  addActionLabel: "Add",
  addActionLink: "/link/"
};

describe("<CategoryListCard />", () => {
  it("displays properly", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryListCard {...listCardProps} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("displays children", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryListCard {...listCardProps}>
          <b>test</b>
        </CategoryListCard>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
