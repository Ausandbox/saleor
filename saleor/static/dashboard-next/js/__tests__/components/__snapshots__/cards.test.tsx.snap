// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryListCard /> displays children 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5"
>
  <div
    className="MuiToolbar-root-31 MuiToolbar-gutters-32"
  >
    <div
      className="Component-title-3"
    >
      <h2
        className="MuiTypography-root-33 MuiTypography-title-39"
      >
        Title
      </h2>
    </div>
    <div
      className="Component-spacer-1"
    />
    <a
      className="MuiButtonBase-root-72 MuiButton-root-57 MuiButton-flatPrimary-59 Component-actions-2"
      href="/link/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-58"
      >
        Add
      </span>
      <span
        className="MuiTouchRipple-root-74"
      />
    </a>
  </div>
  <b>
    test
  </b>
</div>
`;

exports[`<CategoryListCard /> displays properly 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation2-8 MuiPaper-rounded-5"
>
  <div
    className="MuiToolbar-root-31 MuiToolbar-gutters-32"
  >
    <div
      className="Component-title-3"
    >
      <h2
        className="MuiTypography-root-33 MuiTypography-title-39"
      >
        Title
      </h2>
    </div>
    <div
      className="Component-spacer-1"
    />
    <a
      className="MuiButtonBase-root-72 MuiButton-root-57 MuiButton-flatPrimary-59 Component-actions-2"
      href="/link/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-58"
      >
        Add
      </span>
      <span
        className="MuiTouchRipple-root-74"
      />
    </a>
  </div>
</div>
`;
